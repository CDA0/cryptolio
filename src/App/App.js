import React, { Component } from 'react';
import { AppRegistry, StyleSheet, Text, View } from 'react-native';
import TabNavigator from 'react-native-tab-navigator';
import Icon from 'react-native-vector-icons/FontAwesome';
import { Dimensions } from 'react-native';

const deviceW = Dimensions.get('window').width;

const basePx = 375;

function px2dp(px) {
  return px * deviceW / basePx;
}

class Home extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Home</Text>
      </View>
    );
  }
}

class Coins extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Coins</Text>
      </View>
    );
  }
}

class Settings extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Settings</Text>
      </View>
    );
  }
}

export default class TabDemo extends Component {
  state = {
    selectedTab: 'home',
  };

  render() {
    return (
      <TabNavigator style={styles.container}>
        <TabNavigator.Item
          selected={this.state.selectedTab === 'home'}
          title="Home"
          selectedTitleStyle={{ color: '#3496f0' }}
          renderIcon={() => <Icon name="home" size={px2dp(22)} color="#666" />}
          renderSelectedIcon={() => (
            <Icon name="home" size={px2dp(22)} color="#3496f0" />
          )}
          // badgeText="1"
          onPress={() => this.setState({ selectedTab: 'home' })}
        >
          <Home />
        </TabNavigator.Item>
        <TabNavigator.Item
          selected={this.state.selectedTab === 'coins'}
          title="Coins"
          selectedTitleStyle={{ color: '#3496f0' }}
          renderIcon={() => (
            <Icon name="briefcase" size={px2dp(22)} color="#666" />
          )}
          renderSelectedIcon={() => (
            <Icon name="briefcase" size={px2dp(22)} color="#3496f0" />
          )}
          onPress={() => this.setState({ selectedTab: 'coins' })}
        >
          <Coins />
        </TabNavigator.Item>
        <TabNavigator.Item
          selected={this.state.selectedTab === 'settings'}
          title="Settings"
          selectedTitleStyle={{ color: '#3496f0' }}
          renderIcon={() => <Icon name="cog" size={px2dp(22)} color="#666" />}
          renderSelectedIcon={() => (
            <Icon name="cog" size={px2dp(22)} color="#3496f0" />
          )}
          onPress={() => this.setState({ selectedTab: 'settings' })}
        >
          <Settings />
        </TabNavigator.Item>
      </TabNavigator>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

AppRegistry.registerComponent('TabDemo', () => TabDemo);
