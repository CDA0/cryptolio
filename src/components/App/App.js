import React, { Component } from 'react';
import TabNavigator from 'react-native-tab-navigator';

import HomeScreen from '../HomeScreen/HomeScreen';
import CoinsScreen from '../CoinsScreen/CoinsScreen';
import SettingsScreen from '../SettingsScreen/SettingsScreen';
import TabIcon from '../TabIcon/TabIcon';
import { LIGHT_BLUE } from '../../services/colors';

class Cryptolio extends Component {
  state = {
    selectedTab: 'home',
  };

  render() {
    const { selectedTab } = this.state;

    return (
      <TabNavigator>
        <TabNavigator.Item
          selected={selectedTab === 'home'}
          title="Home"
          selectedTitleStyle={{ color: LIGHT_BLUE }}
          renderIcon={() => <TabIcon name="home" />}
          renderSelectedIcon={() => <TabIcon name="home" selected />}
          // badgeText="1"
          onPress={() => this.setState({ selectedTab: 'home' })}
        >
          <HomeScreen />
        </TabNavigator.Item>

        <TabNavigator.Item
          selected={selectedTab === 'coins'}
          title="Coins"
          selectedTitleStyle={{ color: LIGHT_BLUE }}
          renderIcon={() => <TabIcon name="briefcase" />}
          renderSelectedIcon={() => <TabIcon name="briefcase" selected />}
          onPress={() => this.setState({ selectedTab: 'coins' })}
        >
          <CoinsScreen />
        </TabNavigator.Item>

        <TabNavigator.Item
          selected={selectedTab === 'settings'}
          title="Settings"
          selectedTitleStyle={{ color: LIGHT_BLUE }}
          renderIcon={() => <TabIcon name="cog" />}
          renderSelectedIcon={() => <TabIcon name="cog" selected />}
          onPress={() => this.setState({ selectedTab: 'settings' })}
        >
          <SettingsScreen />
        </TabNavigator.Item>
      </TabNavigator>
    );
  }
}

// const styles = StyleSheet.create({
//   instructions: {
//     textAlign: 'center',
//     color: '#333333',
//     marginBottom: 5,
//   },
// });

export default Cryptolio;
