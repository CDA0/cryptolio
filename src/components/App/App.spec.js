import 'react-native';
import React from 'react';
import { App } from './App';

// Note: test renderer must be required after react-native.
import ShallowRenderer from 'react-test-renderer/shallow';

const getShallowRenderer = toRender => {
  const renderer = new ShallowRenderer();
  renderer.render(toRender);
  return renderer;
};

const getShallowOutput = toRender =>
  getShallowRenderer(toRender).getRenderOutput();

it('renders', () => {
  const rendered = getShallowOutput(<App />);

  // expect(rendered).toMatchSnapshot();
  expect(rendered).not.toBe(false);
});
