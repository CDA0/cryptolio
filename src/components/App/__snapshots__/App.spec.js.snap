// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<TabNavigator>
  <TabNavigatorItem
    onPress={[Function]}
    renderIcon={[Function]}
    renderSelectedIcon={[Function]}
    selected={true}
    selectedTitleStyle={
      Object {
        "color": "#3496f0",
      }
    }
    title="Home"
  >
    <HomeScreen />
  </TabNavigatorItem>
  <TabNavigatorItem
    onPress={[Function]}
    renderIcon={[Function]}
    renderSelectedIcon={[Function]}
    selected={false}
    selectedTitleStyle={
      Object {
        "color": "#3496f0",
      }
    }
    title="Coins"
  >
    <CoinsScreen />
  </TabNavigatorItem>
  <TabNavigatorItem
    onPress={[Function]}
    renderIcon={[Function]}
    renderSelectedIcon={[Function]}
    selected={false}
    selectedTitleStyle={
      Object {
        "color": "#3496f0",
      }
    }
    title="Settings"
  >
    <SettingsScreen />
  </TabNavigatorItem>
</TabNavigator>
`;
